### PR TITLE
Update head more timely before

### PR DESCRIPTION
**Pull request by terencechain**
# Description

Move `s.cfg.BeaconDB.SaveJustifiedCheckpoint` further down. Checkpoint in DB is only used at start up or eth2.0 api get state root. This will shred off another 50ms to update head timely

# Data
Before
Screen Shot 2021-09-21 at 9 52 38 AM
After
Screen Shot 2021-09-22 at 11 32 35 AM

### CODE REVIEW

**Comment by terencechain** [comment]
I found one issue at skipping `s.savePostStateInfo` before get head. Get head requires state summary, state and block in DB. We can get around it with the below but it gets ugly. I dont think it's worth it, I rather fix it in another PR via general refactor


```go
// This caches head faster without head block and state stored in DB
func (s *Service) saveHeadFast(ctx context.Context, headRoot [32]byte, headState state.BeaconState, headBlock block.SignedBeaconBlock) error {
	ctx, span := trace.StartSpan(ctx, "blockChain.saveHeadFast")
	defer span.End()


	s.setHead(headRoot, headBlock, headState)
	if err := s.cfg.BeaconDB.SaveHeadBlockRoot(ctx, headRoot); err != nil {
		return errors.Wrap(err, "could not save head root in DB")
	}
```

**Review by rkapka — APPROVED** [review]

File: beacon-chain/blockchain/process_block_helpers.go [hunk 1]
```diff
@@ -212,7 +212,7 @@ func (s *Service) updateJustified(ctx context.Context, state state.ReadOnlyBeaco
 		}
 	}
 
-	return s.cfg.BeaconDB.SaveJustifiedCheckpoint(ctx, cpt)
+	return nil
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/process_block_helpers.go:215]: This makes me so nervous that no test fails for updateJustified.

File: beacon-chain/blockchain/process_block.go [hunk 2]
```diff
@@ -155,6 +141,27 @@ func (s *Service) onBlock(ctx context.Context, signed block.SignedBeaconBlock, b
...
+			}
+		}()
+	}
+
+	// Save justified check point to db.
+	if postState.CurrentJustifiedCheckpoint().Epoch > s.justifiedCheckpt.Epoch {
+		if err := s.cfg.BeaconDB.SaveJustifiedCheckpoint(ctx, postState.CurrentJustifiedCheckpoint()); err != nil {
+			return err
+		}
+	}
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/process_block.go:163]: Can we add some kind of test to support this moved logic?

File: beacon-chain/blockchain/process_block_helpers.go (hunk 1 above)
Comment (prestonvanloon) [reply beacon-chain/blockchain/process_block_helpers.go:215]: Why is this removed anyway?

**Pull request merged**